yarl/_path.py

```diff
@@ -39,3 +40,24 @@ def normalize_path(path: str) -> str:
 
     segments = path.split("/")
     return prefix + "/".join(normalize_path_segments(segments))
+
+
+def relative_path(path: str, start: str) -> str:
+    """A wrapper over os.path.relpath()"""
+
+    if not path:
```

Comment:
Unfortunately, this will not work with `os.path.relpath()`. Anyway, I've added a check for absolute and relative paths, so I don't think there will be any more problems with this 🤷‍♂️

Reply:
You'll need to use https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.relative_to instead

Here are the equivalent functions
https://docs.python.org/3/library/pathlib.html#corresponding-tools

Reply:
This sounds like a good idea, but `os.path.relpath()` and `PurePath.relative_to()` behave differently. For example, `PurePath("/path/to").relative_to("/path/from")` raises a `ValueError`: '/path/to' is not in the subpath of '/path/from'.
And `os.path.relpath("/path/to", "/path/from")` will just return `../to`.
Would you still like to replace `os.path.relpath()` with `PurePath.relative_to()`?

Reply:
It looks like `relpath` has a lot of potential calls into [`abspath`](https://github.com/python/cpython/blob/44f841f01af0fb038e142a07f15eda1ecdd5b08a/Lib/posixpath.py#L376) which will call `getcwd`

I think you could build something based on [`relpath`](https://github.com/python/cpython/blob/44f841f01af0fb038e142a07f15eda1ecdd5b08a/Lib/posixpath.py#L497) which would guarantee no I/O happens.

Reply:
Sorry, I was a bit tired, so I didn't notice at first that `PurePath.relative_to()` takes a `walk_up` argument. Now everything works as it should, so I can completely get rid of `os.path` in my code! 🎉

Reply:
I've got a new problem. The `walk_up` parameter was added only in Python 3.12 🤦‍♂️. I will try to do something with it...

Reply:
I made it work with Python 3.9 🎉